JavaFiddle-ejb/src/main/java/ru/javafiddle/core/ejb/ProjectBean.java

```diff
@@ -1,211 +1,282 @@
+            return null;
+        }
+        return p;
+    }
 
-        Group group = (Group)em.createQuery("SELECT g FROM Group g WHERE g.groupId=:groupid")
-                .setParameter("groupid", groupId)
-                .getSingleResult();
+    public Project getProjectByName(String projectName) {
 
-        NavigableMap<String, String> m = new TreeMap<>();
-        return group;
+        Project p;
+        try {
+            p = (Project) em.createQuery("SELECT p FROM Project p WHERE p.projectName =:projectname")
+                    .setParameter("projectname", projectName)
+                    .getSingleResult();
+        }catch (NoResultException noResult) {
+            logger.log(Level.WARNING, "NO RESULT IN QUERY", noResult);
+            return null;
+        }
+        return p;
     }
 
-    private void createGroup(int groupId) {
-        Group g = new Group();
 
-        g.setGroupName("default");
-        g.setGroupId(groupId);
+    public int getGroupId(String projectHash) {
 
+        Project p = getProject(projectHash);
 
+        int groupId = p.getGroup().getGroupId();
 
+        return groupId;
 
+    }
 
 
-    }
 
 
+    public void deleteProject(String projectHash){
+
+        Project project = getProject(projectHash);
+
+        em.remove(project);
 
-    public String getHash(String password) throws NoSuchAlgorithmException, UnsupportedEncodingException {
-        MessageDigest digest = MessageDigest.getInstance("SHA-1");
-        digest.reset();
-        byte[] input = digest.digest(password.getBytes("UTF-8"));
 
-        String res = new String(input, "UTF-8");
-        return res;
     }
 
-    //!TODO
-    public int getGroupId(String projectHash) {
-        return -1;
+    public Project changeProjectName(String projectHash, String newProjectName) {
+
+        Project project;
+        project = getProject(projectHash);
+
+        if (project == null) {
+
+            logger.log(Level.SEVERE, "NO RESULT IN QUERY");
+            return null;
+        }
+        project.setProjectName(newProjectName);
+
+        em.persist(project);
+        return getProject(projectHash);
     }
 
-    //!TODO
-    public void deleteProject(String projectHash){
+
+    @TransactionAttribute(TransactionAttributeType.REQUIRED)
+    public Group addProject(Group group, Project project) {
+
+        if (group.getProjects() == null) {
+
+            List<Project> p = new LinkedList<>();
+            p.add(project);
+            group.setProjects(p);
+        }
+        List<Project> p = group.getProjects();
+        p.add(project);
+        group.setProjects(p);
+        em.persist(group);
+
+        return group;
 
     }
 
-    //!TODO
-    public void changeProjectName(String projectHash, String newProjectName) {
+    @TransactionAttribute(TransactionAttributeType.REQUIRES_NEW)
+    public Project createCopy(String hash) throws UnsupportedEncodingException, NoSuchAlgorithmException {
```

Comment:
So, if we correct it everywhere , all services should also be corrected and beans and device won't be separated anymore.

23 марта 2016 г., в 3:03, alexander1leonov <notifications@github.com> написал(а):

> In JavaFiddle-ejb/src/main/java/ru/javafiddle/core/ejb/ProjectBean.java:
> 
> >  
> >      }
> >  
> > -    //!TODO
> > -    public void changeProjectName(String projectHash, String newProjectName) {
> > +    @TransactionAttribute(TransactionAttributeType.REQUIRES_NEW)
> > +    public Project createCopy(String hash) throws UnsupportedEncodingException, NoSuchAlgorithmException {
> Please consider using Project entity as input param
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>